src/zh/yidan/src/eu/kanade/tachiyomi/extension/zh/yidan/Yidan.kt

```diff
@@ -1,135 +1,348 @@
+        headers,
+        ComicFetchRequest("29", page, PAGE_SIZE).toJsonRequestBody(),
+    )
 
     override fun popularMangaParse(response: Response): MangasPage {
-        val listing: ListingDto = response.parseAs()
-        val mangas = listing.list.map { it.toSManga(baseUrl) }
-        val hasNextPage = run {
-            val url = response.request.url
-            val pageSize = url.queryParameter("pageSize")!!.toInt()
-            val pageNumber = url.queryParameter("pageNo")!!.toInt()
-            pageSize * pageNumber < listing.totalCount
-        }
-        return MangasPage(mangas, hasNextPage)
+        val records = response.parseAs<CommonResponse<RecordResult>>().result.records
+        return createMangasPage(records)
     }
 
-    override fun latestUpdatesRequest(page: Int) =
-        GET("$baseUrl/prod-api/app-api/vv/mh-list/page?mhcate=4&pageSize=50&pageNo=$page", headers)
+    override fun latestUpdatesRequest(page: Int) = POST(
+        "$baseUrl/api/getByComicByRow",
+        headers,
+        ComicFetchRequest("34", page, PAGE_SIZE).toJsonRequestBody(),
+    )
 
     override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
 
+    private fun searchByKeyword(page: Int, query: String): Request {
+        return POST(
+            "$apiUrl/api/searchNovel",
+            headers,
+            KeywordSearchRequest(query).toJsonRequestBody(),
+        )
+    }
+
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val url = "$baseUrl/prod-api/app-api/vv/mh-list/page".toHttpUrl().newBuilder()
-            .apply { if (query.isNotBlank()) addQueryParameter("word", query) }
-            .apply { parseFilters(filters, this) }
-            .addEncodedQueryParameter("pageSize", "50")
-            .addEncodedQueryParameter("pageNo", page.toString())
-            .build()
-        return Request.Builder().url(url).headers(headers).build()
+        if (query.isNotEmpty()) {
+            return searchByKeyword(page, query)
+        }
+        return POST(
+            "$apiUrl/api/getByComicCategoryId",
+            headers,
+            FilterRequest(
+                page = page,
+                limit = PAGE_SIZE,
+                categoryId = filters.firstInstance<CategoryFilter>().selected,
+                orderType = filters.firstInstance<SortFilter>().selected,
+                overType = filters.firstInstance<StatusFilter>().selected,
+            ).toJsonRequestBody(),
+        )
     }
 
-    override fun searchMangaParse(response: Response) = popularMangaParse(response)
+    override fun searchMangaParse(response: Response): MangasPage {
+        val searchByKeyword = response.request.url.toString().contains("searchNovel")
+        val records = if (searchByKeyword) {
+            response.parseAs<CommonResponse<List<Record>>>().result
+        } else {
+            response.parseAs<CommonResponse<FilterResult>>().result.list
+        }
```

Comment:
Consider something like:
```suggestion
        val records = when {
            searchByKeyword -> response.parseAs<CommonResponse<List<Record>>>().result
            else -> response.parseAs<CommonResponse<FilterResult>>().result.list
        }
```